### PR TITLE
Add cache-control headers to implementation API

### PR DESCRIPTION
**Pull request by mythmon**
This will make the CDN happy. And really everyone who wants this service to be fast fast fast. I confirmed that Cloudfront (the CDN we will be using) will pay attention tot his cache header.

@Osmose r?

### CODE REVIEW

File: normandy/recipes/api/views.py [hunk 1]
```diff
@@ -54,4 +54,9 @@ def retrieve(self, request, name, impl_hash):
         if impl_hash != action.implementation_hash:
             raise NotFound('Hash does not match current stored action.')
 
-        return Response(action.implementation)
+        max_age = 60 * 60 * 24 * 365  # one year in seconds
```
Comment (Osmose) [inline normandy/recipes/api/views.py]: Setting?

File: normandy/recipes/api/views.py [hunk 2]
```diff
@@ -54,4 +54,9 @@ def retrieve(self, request, name, impl_hash):
...
+        headers = {
```
Comment (Osmose) [inline normandy/recipes/api/views.py]: Would the `cache_control` decorator work here instead?

**Comment by Osmose** [comment]
r+wc

File: normandy/recipes/api/views.py (hunk 2 above)
Comment (mythmon) [inline normandy/recipes/api/views.py]: TIL

**Pull request merged**